dxr/query.py

```diff
@@ -126,52 +161,39 @@ def group_filters_by_name(predicate):
         is_line_query = any(f.domain == LINE for f in
                             chain.from_iterable(filters))
 
-        # An ORed-together ball for each term's filters, omitting filters that
-        # punt by returning {} and ors that contain nothing but punts:
-        ors = filter(None, [filter(None, (f.filter() for f in term))
-                            for term in filters])
-        ors = [{'or': x} for x in ors]
-
-        if not is_line_query:
-            # Don't show folders yet in search results. I don't think the JS
-            # is able to handle them.
-            ors.append({'term': {'is_folder': False}})
-            # Filter out all FILE docs who are links.
-            ors.append({'not': {'exists': {'field': 'link'}}})
-
-        if ors:
-            query = {
-                'filtered': {
-                    'query': {
-                        'match_all': {}
-                    },
-                    'filter': {
-                        'and': ors
-                    }
-                }
+        return self._do_search_and_highlight(_make_es_query(filters, is_line_query),
+                                             filters, is_line_query,
+                                             offset, limit)
+
+    # Test: If var-ref (or any structural query) returns 2 refs on one line, they should both get highlit.
+
+    def promoted_paths(self, term, promote_limit=5):
+        """Return a tuple (total number of path results, promoted paths, querystring),
+        where the number of promoted paths is no more than promote_limit."""
+
+        # Filter out the FILE domain filters, and add our own path filter at the end.
+        file_filters = [[f for f in named_filters if f.domain == FILE] for named_filters in
+                        self._filters] + [[PathFilter(term, self.enabled_plugins)]]
+        filtered_query = _make_es_query(file_filters, False)
+        path_query = {'constant_score': {'query': filtered_query, 'boost': 0.5}}
```

Comment:
It would have helped me if the names of `path_query` and `exact_query` contrasted with each other, like maybe `segment_query` and `substring_query` or something.